Sources/StreamChat/WebSocketClient/EventMiddlewares/UserChannelBanEventsMiddleware.swift

```diff
@@ -18,41 +16,33 @@ struct UserChannelBanEventsMiddleware<ExtraData: ExtraDataTypes>: EventMiddlewar
             completion(event)
             return
         }
-
-        switch event {
-        case let userBannedEvent as UserBannedEvent:
-            database.write { session in
+        
+        do {
+            switch event {
+            case let userBannedEvent as UserBannedEvent:
                 guard let memberDTO = session.member(userId: userBannedEvent.userId, cid: cid) else {
                     throw ClientError.MemberDoesNotExist(userId: userBannedEvent.userId, cid: cid)
                 }
-
+                
                 memberDTO.isBanned = true
                 memberDTO.banExpiresAt = userBannedEvent.expiredAt
-            } completion: { error in
-                if let error = error {
-                    log.error("Failed to write `UserBannedEvent` updates to the DB: \(error)")
-                }
-                completion(event)
-            }
-
-        case let userUnbannedEvent as UserUnbannedEvent:
-            database.write { session in
+                
+            case let userUnbannedEvent as UserUnbannedEvent:
                 guard let memberDTO = session.member(userId: userUnbannedEvent.userId, cid: cid) else {
                     throw ClientError.MemberDoesNotExist(userId: userUnbannedEvent.userId, cid: cid)
                 }
-
+                
                 memberDTO.isBanned = false
                 memberDTO.banExpiresAt = nil
-            } completion: { error in
-                if let error = error {
-                    log.error("Failed to write `UserUnbannedEvent` updates to the DB: \(error)")
-                }
-                completion(event)
+                
+            default:
+                break
```

Comment:
maybe add here some assertion or some log that this is unhandled case?

Reply:
Hmm 🤔 The only requirement for the incoming event to be processed is to have a `payload` with `cid`. There are many events that pass these rules besides `UserBannedEvent/UserUnbannedEvent` -> traping into `default` is not a failure but just a normal scenario -> I'd not put any log there.